### ISSUE TITLE
Question about Potential Label Leakage in Using preprocess_logits_for_metrics for Evaluation

### ISSUE THREAD

**Issue by YRYangang** [sunlin-ai/FreePRM#1]
Hi authors,

First of all, thanks for open-sourcing.

While reading through the code, I noticed that the function `preprocess_logits_for_metrics` uses labels to construct the probability of being correct/incorrect:

https://github.com/sunlin-ai/FreePRM/blob/2ee410aa2d7dd40171a18cbdbe81da068c8c02af/train_prm.py#L240-L243

This is fine for training, but such constructed probability is also used for evaluation:

https://github.com/sunlin-ai/FreePRM/blob/2ee410aa2d7dd40171a18cbdbe81da068c8c02af/train_prm.py#L200

Since the calculation of probability is directly ground-truth-related, I do think there is a label leakage problem here, and perphaps the correct way to do eval is using the original logits shape: `(B, S, 3)` and compare the value in the "correct dimension" and "incorrect dimension", e.g., `correct if prob[..., 0] > prob[..., 1]`.

I might be wrong but could you clarify if uch an issue exist? Thank you.
